src/linuxX64Main/kotlin/mu/internal/Formatter.kt

```diff
@@ -0,0 +1,92 @@
+package mu.internal
+
+import kotlinx.cinterop.*
+import platform.posix.*
+
+
+abstract class AFormatter {
+    abstract fun makeFormat(): String
+    override fun toString(): String {
+        return makeFormat()
+    }
+}
+
+/**
+ * Gets a date struct, [upstream docs](http://www.cplusplus.com/reference/ctime/tm/)
+ */
+fun getDateStruct() = memScoped {
+    val t = alloc<time_tVar>()
```

Comment:
`val t = cValue<time_tVar> { value = time(null) }` is an option. Then you don't need `memScoped`.

Reply:
I had seen a similar snippet at one point.

Using

```
val t = cValue<time_tVar> { value = time(null) }
// Gave
Type argument is not within its bounds: should be subtype of 'CStructVar'
// For <time_tVar>
```

Reply:
Ah! Nvm then.